src/main/java/lib/motors/TalonFXSim.java

```diff
@@ -0,0 +1,110 @@
+package lib.motors;
+
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.controls.*;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import lib.math.differential.Derivative;
+import lib.units.Units;
+
+public class TalonFXSim {
+
+    private final DCMotorSim motorSim;
+    private final Derivative acceleration = new Derivative();
+    private PIDController controller = null;
+    private ProfiledPIDController profiledController = null;
+    private double lastTimestampSeconds = 0;
+
+    private double voltageRequest;
+
+    public TalonFXSim(int numMotors, double gearing, double jKgMetersSquared) {
+        DCMotor motor = DCMotor.getFalcon500(numMotors);
+
+        motorSim = new DCMotorSim(motor, gearing, jKgMetersSquared);
+    }
+
+    public void update(double timestampSeconds) {
+        motorSim.update(timestampSeconds - lastTimestampSeconds);
+        acceleration.update(getVelocity(), timestampSeconds);
+
+        lastTimestampSeconds = timestampSeconds;
+    }
+
+    public TalonFXSim setController(PIDController controller) {
+        this.controller = controller;
+        return this;
+    }
+
+    public TalonFXSim setProfiledController(ProfiledPIDController profiledController) {
+        this.profiledController = profiledController;
+        return this;
+    }
+
+    public StatusCode setControl(DutyCycleOut request) {
+        return this.setControl(new VoltageOut(request.Output * 12));
```

Comment:
Consider using WPILib's [`BatterySim`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/simulation/BatterySim.html) as recommended in the "Important" note [here](https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/simulation/simulation-intro.html#inputs-and-outputs). This is relevant to `SparkMaxSim` too.

Reply:
Maybe next off-season.